src/core/service.c

```diff
@@ -2006,6 +2006,26 @@ static void service_kill_control_process(Service *s) {
         }
 }
 
+static bool service_adverse_to_leftover_processes(Service *s) {
```

Comment:
given that when you call this you need to initialize `r` to something maybe just return an `int` here, and then return an error like `-EBUSY` when we refuse to proceed like this, and 0 otherwise?